sdk/storage/inc/datalake/directory_client.hpp

```diff
@@ -0,0 +1,157 @@
+    // *             files and directories, PathSetAccessControlRecursiveMode::Modify modifies one
+    // or
+    // *             more POSIX access control rights  that pre-exist on files and directories,
+    // *             PathSetAccessControlRecursiveMode::Remove removes one or more POSIX access
+    // *             control rights that were present earlier on files and directories
+    // * @param acls Sets POSIX access control rights on files and directories. Each access control
+    // *             entry (ACE) consists of a scope, a type, a user or group identifier, and
+    // *             permissions.
+    // * @param options Optional parameters to set an access control recursively to the resource the
+    // *                directory points to.
+    // * @return Azure::Core::Response<DirectorySetAccessControlRecursiveInfo>
+    // * @remark This request is sent to dfs endpoint.
+    // */
+    // Azure::Core::Response<DirectorySetAccessControlRecursiveInfo> SetAccessControlRecursive(
+    //    PathSetAccessControlRecursiveMode mode,
+    //    std::vector<Acl> acls,
+    //    const SetAccessControlRecursiveOptions& options = SetAccessControlRecursiveOptions())
+    //    const;
+
+    /**
+     * @brief Create a directory. By default, the destination is overwritten and
+     *        if the destination already exists and has a lease the lease is broken.
+     * @param options Optional parameters to create the directory the path points to.
+     * @return Azure::Core::Response<DirectoryInfo>
+     * @remark This request is sent to dfs endpoint.
+     */
+    Azure::Core::Response<DirectoryInfo> Create(
+        const DirectoryCreateOptions& options = DirectoryCreateOptions()) const
+    {
+      return PathClient::Create(PathResourceType::Directory, options);
+    }
+
+    /**
+     * @brief Renames a directory. By default, the destination is overwritten and
+     *        if the destination already exists and has a lease the lease is broken.
+     * @param destinationDirectoryPath The destinationPath this current directory is renaming to.
+     * @param options Optional parameters to rename a resource to the resource the destination
+     * directory points to.
+     * @return Azure::Core::Response<DirectoryRenameInfo>
+     * @remark This will change the URL the client is pointing to.
+     * @remark This request is sent to dfs endpoint.
+     */
+    Azure::Core::Response<DirectoryRenameInfo> Rename(
+        const std::string& destinationDirectoryPath,
+        const DirectoryRenameOptions& options = DirectoryRenameOptions());
+
+    /**
+     * @brief Deletes the directory.
+     * @param options Optional parameters to delete the directory the path points to.
+     * @return Azure::Core::Response<DirectoryDeleteResponse>
+     * @remark This request is sent to dfs endpoint.
+     */
+    Azure::Core::Response<DirectoryDeleteInfo> Delete(
```

Comment:
Please confirm with @JeffreyRichter that he's okay this this naming rule.

Reply:
Some minor changes but seems to be OK.